nemo/collections/nlp/models/language_modeling/megatron_gpt_prompt_learning_model.py

```diff
@@ -344,7 +377,34 @@ def validation_step(self, batch, batch_idx):
         if loss_mean.item == 0.0:
             loss_mean = []
 
-        return loss_mean
+        if self.cfg.get('report_validation_metric', False):
+            preds_text, labels_text = [], []
+            input_ids, labels, loss_mask, position_ids, attention_mask, taskname_ids = batch
+            input_lenghts = torch.cuda.LongTensor(
+                [input.shape[0] - self.length_params['max_length'] for input in input_ids]
+            )
+
+            res = megatron_gpt_generate(
+                self.cuda(),
+                (input_ids, torch.cuda.LongTensor(input_lenghts)),
+                self.tokenizer,
+                self.length_params,
+                self.sampling_params,
+                task_ids=taskname_ids,
+            )
+
+            for pred, label in zip(res['token_ids'], labels):
+                pred = self.tokenizer.ids_to_text(pred)
+                label = self.tokenizer.ids_to_text(label)
```

Comment:
If you can, please leave a comment that mentions that ids_to_text strips special tokens by itself, in case, it's not obvious from the name itself.